src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -188,6 +190,42 @@ internal void AppendMetadata(MethodBodyStreamEncoder methodBodyEncoder)
             }
         }
 
+        private void WriteProperties(TypeBuilderImpl typeBuilder)
+        {
+            if (typeBuilder._propertyDefinitions.Count > 0)
+            {
+                AddPropertyMap(typeBuilder._handle, typeBuilder._firstPropertyToken);
+                foreach (PropertyBuilderImpl property in typeBuilder._propertyDefinitions)
+                {
+                    PropertyDefinitionHandle propertyHandle = AddPropertyDefinition(property, MetadataSignatureHelper.PropertySignatureEncoder(property, this));
+                    WriteCustomAttributes(property._customAttributes, propertyHandle);
+
+                    if (property.GetMethod is MethodBuilderImpl gMb)
+                    {
+                        AddMethodSemantics(propertyHandle, MethodSemanticsAttributes.Getter, gMb._handle);
+                    }
+
+                    if (property.SetMethod is MethodBuilderImpl sMb)
+                    {
+                        AddMethodSemantics(propertyHandle, MethodSemanticsAttributes.Setter, sMb._handle);
+                    }
+
+                    if (property._otherMethods != null)
+                    {
+                        foreach (MethodBuilderImpl method in property._otherMethods)
```

Comment:
Do we have tests for other property methods? IL-only tests?

Reply:
I have only this: https://github.com/dotnet/runtime/blob/dece90dd90b5e9197e9089ddb3b21d07f0c60422/src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySavePropertyBuilderTests.cs#L56-L58

In the generated assembly it was added just like a normal method, was not sure what else to verify

Reply:
I assume this was for ".other" methods which are different than ".get" and ".set" (ECMA allows methods other than get\set)

Reply:
Yes, it is the "other" method